recipes/gdal/all/patches/3.1.x/fix-autotools-3.1.4.patch

```diff
@@ -0,0 +1,281 @@
+ 
+-CONFIG_LIBS	=	$(GDAL_ROOT)/$(LIBGDAL)
++CONFIG_LIBS	=	$(GDAL_ROOT)/$(LIBGDAL) $(LIBS)
+ ifeq ($(MACOSX_FRAMEWORK),yes)
+ CONFIG_LIBS_INS	=	-L$(INST_LIB)/unix/lib -lgdal
+ else
+--- a/configure.ac
++++ b/configure.ac
+@@ -1214,6 +1214,7 @@ dnl ---------------------------------------------------------------------------
+ dnl PROJ.6 related stuff
+ dnl ---------------------------------------------------------------------------
+ 
++PKG_PROG_PKG_CONFIG([0.21])
+ AC_MSG_CHECKING([for PROJ >= 6 library])
+ 
+ AC_ARG_WITH(proj,[  --with-proj=ARG Compile with PROJ.x (ARG=yes or path)],,)
+@@ -1229,11 +1230,7 @@ if test "x$with_proj" = "xno" ; then
+ else
+ 
+   if test "x$with_proj" = "xyes" -o "x$with_proj" = "x"; then
+-    ORIG_LIBS="$LIBS"
+-    LIBS="-lproj $ORIG_LIBS"
+-    AC_LANG_PUSH([C++])
+-    AC_CHECK_LIB(proj,proj_create_from_wkt,PROJ_FOUND=yes,PROJ_FOUND=no,)
+-    AC_LANG_POP([C++])
++    PROJ_FOUND=yes
+     if test "$PROJ_FOUND" = "no"; then
+         AC_LANG_PUSH([C++])
+         AC_CHECK_LIB(proj,internal_proj_create_from_wkt,PROJ_FOUND=yes,PROJ_FOUND=no,)
+@@ -1876,7 +1873,6 @@ else
+ 
+   DDS_SETTING=yes
+   CRUNCHDIR="$with_dds"
+-  LIBS="-L$with_dds/lib/ -lcrunch $LIBS"
+   echo "using libcrunch from $with_dds."
+ 
+ fi
+@@ -2012,7 +2008,7 @@ else
+   AC_MSG_RESULT([using libtiff from ${with_libtiff}.])
+ 
+   dnl Check for the BigTIFF enabled library (libtiff >= 4.0)
+-  AC_CHECK_LIB(tiff,TIFFScanlineSize64,HAVE_BIGTIFF=yes,HAVE_BIGTIFF=no,)
++  AC_CHECK_LIB(tiff,TIFFScanlineSize64,HAVE_BIGTIFF=yes,HAVE_BIGTIFF=no,$LIBS)
+ 
+ fi
+ 
+@@ -2082,7 +2078,6 @@ else
+ 
+   dnl We now require libgeotiff 1.5.0
+   dnl first check if $with_geotiff/lib has the library:
+-  AC_CHECK_LIB(geotiff,GTIFAttachPROJContext,GEOTIFF_SETTING=external,GEOTIFF_SETTING=not_found,-L$with_geotiff/lib)
+ 
+   if test $GEOTIFF_SETTING = "external" ; then
+     LIBS="-L$with_geotiff/lib -lgeotiff $LIBS"
+@@ -2650,6 +2645,10 @@ if test x"$with_hdf4" = x"no" ; then
+ 
+   echo "hdf4 support disabled."
+ 
++elif test x"$with_hdf4" = x"yes" ; then
++  PKG_CHECK_MODULES([HDF4], [hdf], [HAVE_HDF4=yes], [HAVE_HDF4=no])
++  PKG_CHECK_MODULES([HDF4], [hdf >= 4.2.5], [HDF4_HAS_MAXOPENFILES=yes], [HDF4_HAS_MAXOPENFILES=no])
++  HDF4_INCLUDE="$HDF4_CFLAGS"
+ else
+ 
+   if test x"$with_hdf4" = x"yes" -o x"$with_hdf4" = x"" ; then
+@@ -2780,7 +2779,6 @@ elif test "$with_hdf5" = "yes" -o "$with_hdf5" = "" ; then
+     # Test that the package found is for the right architecture
+     saved_LIBS="$LIBS"
+     LIBS="$HDF5_LIBS"
+-    AC_CHECK_LIB(hdf5,H5Fopen, [HAVE_HDF5=yes], [HAVE_HDF5=no])
+     LIBS="$saved_LIBS"
+ 
+     if test "$HAVE_HDF5" = "yes"; then
+@@ -2849,21 +2847,21 @@ KEA_CONFIG=no
+ 
+ AC_ARG_WITH(kea,[  --with-kea[=ARG]      Include kealib (ARG=path to kea-config) [[default=yes]]],,)
+ 
++AC_MSG_CHECKING([for kea])
+ if test "$with_kea" = "yes" -o "x$with_kea" = "x" ; then
+-  AC_PATH_PROG(KEA_CONFIG, kea-config, no)
++  HAVE_KEA=yes
```

Comment:
Does kea also need a `PKG_CHECK_MODULES`?

Reply:
`LIBS` and flags variables are already properly set by `AutoToolsBuildEnvironment`, so in this context `PKG_CHECK_MODULES` is useless. It simplify the patch.